### ISSUE TITLE
Cannot run Lucida

### ISSUE THREAD

**Issue by k0105** [claritylab/lucida#116]
First of all, a heads-up: Lucida won't compile under GCC 5 (and hence Ubuntu 16.04 LTS in default configuration) due to ABI changes and it doesn't work with Firefox (you state to only use Chrome, but it is still a nuisance).

Pulling the Docker containers does not work for me: `docker-compose up` from Lucida's main directory pulls the images, but fails afterwards with:
```
lucida_services_1  | bash: line 0: cd: tools: No such file or directory
lucida_lucida_services_1 exited with code 1
```

Using the tutorial download from lucida.ai does work, but it is difficult to work inside docker containers. Also, if I enter the container via `docker run -it --entrypoint="/bin/bash" -p 8081:8081 -p 8090:8090 claritylab/lucida:tutorial`, add a `System.out.println("X");` to `QAServiceHandler.java` and then commit the changes via `docker commit -m "This will fail." e7b40fd771b5 claritylab/lucida:tutorial` the system will start inside the container with supervisord, but fail when launched via Docker Compose:
```
Recreating lucida_lucida_services_1
Recreating lucida_lucida_asr_1
Attaching to lucida_lucida_services_1, lucida_lucida_asr_1
lucida_services_1 | /bin/bash: /bin/bash: cannot execute binary file
lucida_lucida_services_1 exited with code 126
```

Directly compiling the Github on Ubuntu 14.04 works, but during execution I get:
```
2016-05-13 17:59:19,557 INFO exited: imagematching (exit status 1; not expected)
2016-05-13 17:59:19,557 INFO exited: commandcenter (exit status 1; not expected)
2016-05-13 17:59:19,559 INFO gave up: imagematching entered FATAL state, too many start retries too quickly
2016-05-13 17:59:19,559 INFO gave up: commandcenter entered FATAL state, too many start retries too quickly
2016-05-13 17:59:19,559 INFO exited: speechrecognition (exit status 1; not expected)
2016-05-13 17:59:19,559 INFO gave up: speechrecognition entered FATAL state, too many start retries too quickly
2016-05-13 17:59:19,559 INFO exited: questionanswering (exit status 1; not expected)
2016-05-13 17:59:20,561 INFO gave up: questionanswering entered FATAL state, too many start retries too quickly

```
The original error might deviate a bit, since I did not go over all environment variables again, but when I did, it also failed and in a very similar fashion including looping all the time (supervisord seemed to constantly restart IMM or QA, I vaguely recall).

Bottom line: Most ways to run Lucida break on my system. I can only run the tutorial Docker containers properly and once I change anything they break.

Could you help me out with this? Thanks in advance.

Best wishes,
Joe

PS:
Speech detection performs not too well or at least doesn't understand *me*. What is the easiest way to replace it with Nuance / IBM?

**Comment by yunshengb** [comment]
First of all, we are working on the next generation of Lucida, which will be released by the end of this summer, but should be earlier than that. 

We will have an entirely new command center, a better QA experience, a web front end, a new service, and detailed instructions on how to replace individual components with your own. 

We will test the new version on Ubuntu 16.04, and fix the supervisord issue. 

For now, you may ignore the supervisord way of starting all services and start them manually instead. Make sure to setenvironment variables and start the services in the order of: command center -> ASR, IMM, and QA. 

Thank you for your feedback, and please understand that we are working very hard. Please give us more suggestions on how to improve Lucida!

**Comment by bgrayburn** [comment]
are there to be no changes to this project until the next version? would you accept a pull request to fix the docker stuff? I understand you are working hard, and this is an awesome project, but I'm hoping I can actually start using/developing with it. at the very least if you know things in the repo are broken, prevent many wasted man hours by making it clear that things aren't currently working as described.

**Comment by yunshengb** [comment]
I will accept pull request. Please do so to fix the current version. I am not the guy who created the docker stuff, but I can take care of it. Thanks!

**Comment by cbriard** [comment]
Dear all, I'm trying to install Lucida from the last release on Ubuntu 16.04 LTS and it looks like a nightmare, first with the dependences ... Thrift, python 2.9 ...
Is there some plan to correct at least that release or should we make a SHIFT+DEL on that version ?
Thanks a lot in advance,
Christian
